src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,132 @@
 import React from 'react';
 import './TodoList.scss';
+import classNames from 'classnames';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
+type Props = {
+  todos: Todo[];
+  selectUser: (id: number) => void;
+};
+
+type State = {
+  input: string;
+  option: string;
+};
+
+export class TodoList extends React.Component<Props, State> {
+  state = {
+    input: '',
+    option: 'all',
+  };
+
+  changeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      input: event.target.value,
+    });
+  };
+
+  selectHandler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      option: event.currentTarget.value,
+    });
+  };
+
+  filterTodos = () => {
+    const { todos } = this.props;
+    const { input, option } = this.state;
+    let filteredTodos;
+
+    switch (option) {
+      case 'active':
+        filteredTodos = todos.filter(todo => !todo.completed);
+        break;
+
+      case 'completed':
+        filteredTodos = todos.filter(todo => todo.completed);
+        break;
+
+      default:
+        filteredTodos = [...todos];
+    }
+
+    return filteredTodos.filter(({ title }) => (
+      title.includes(input)
+    ));
+  };
+
+  findCompleted = () => {
+    const { todos } = this.props;
+
+    return todos.filter(({ completed }) => (
+      completed
+    ));
+  };
+
+  render() {
+    const { selectUser } = this.props;
+    const { input, option } = this.state;
+
+    const filteredTodos = this.filterTodos();
+
+    // eslint-disable-next-line no-console
+    console.log(filteredTodos);
+
```

Comment:
remove

Reply:
Don't leave any comments and logs in the production code